deps/rabbit/src/rabbit_queue_consumers.erl

```diff
@@ -22,7 +22,8 @@
 
 -define(QUEUE, lqueue).
 
--define(UNSENT_MESSAGE_LIMIT,          200).
+-define(UNSENT_MESSAGE_LIMIT,
+    application:get_env(rabbit, classic_queue_consumer_unsent_message_limit, 200)).
```

Comment:
Can this be initialized in the `cr` record state, or at least a persistent term? Calling `application:get_env/3` for every batch of deliveries sounds substantially more expensive.

Reply:
It sure can. I actually tried initialising in `cr` state but fetching limit by pattern match per message delivery yielded slower throughput numbers vs lookups in ets via `application:get_env/3`. I'll do more samples. Didn't try persistent term. i'll test all 3 and we opt for fastest approach 👍